executor/oci/mounts.go

```diff
@@ -68,7 +68,7 @@ func withROBind(src, dest string) func(m []specs.Mount) ([]specs.Mount, error) {
 			Destination: dest,
 			Type:        "bind",
 			Source:      src,
-			Options:     []string{"rbind", "ro"},
+			Options:     []string{"nosuid", "noexec", "nodev", "rbind", "ro"},
```

Comment:
Which one  of these new flags is related to the `/etc/resolv.conf` EPERM?

Reply:
Seems it needs all of the flags: "nosuid", "noexec", "nodev"

from the pr of moby https://github.com/moby/moby/pull/35205
> When runc is bind-mounting a particular path "with options", it has to
do so by first creating a bind-mount and the modifying the options of
said bind-mount via remount. However, in a user namespace, there are
restrictions on which flags you can change with a remount (due to
CL_UNPRIVILEGED being set in this instance). Docker historically has
ignored this, and as a result, internal Docker mounts (such as secrets)
haven't worked with --userns-remap. Fix this by preserving
CL_UNPRIVILEGED mount flags when Docker is spawning containers with user
namespaces enabled.

I have found related part from kernel,
kernel locks mount tree
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/fs/namespace.c#n2154

while remount it checks whether it can change locked flags
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/fs/namespace.c#n2594

from the function, it sees all the flags
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/fs/namespace.c#n2469